### PR TITLE
Enable python 2.7

### PR DESCRIPTION
**Pull request by heston**
Use `six` to make this compatible with python 2.7.

### CODE REVIEW

**Comment by heston** [comment]
@thedrow FYI, this fixes python 2.7 builds.

**Comment by thedrow** [comment]
Great. Now we need to avoid installing cffi on PyPy and we're done.

**Comment by heston** [comment]
@nitely How would you feel about merging this?

**Comment by nitely** [comment]
Great job! I was going to review this tomorrow, but I'll do it now since I'm already here :stuck_out_tongue_winking_eye:

File: tests/tests_context.py [hunk 1]
```diff
@@ -236,17 +241,19 @@ def get_exception_message(ctx, script):
 
         # todo: trim source line when too long
         with context.Context(self.vm) as ctx:
-            ctx.run_script(script_oops, identifier='my_file_áéíóú.js')
+            ctx.run_script(script_oops, identifier=u'my_file_áéíóú.js'.encode('utf-8'))
```
Comment (nitely) [inline tests/tests_context.py]: You should not change the tests. If it fails then there is something wrong with the code you are testing. Also, don't add the `u` prefix, import `from __future__ import unicode_literals` instead.

File: tests/tests_context.py [hunk 2]
```diff
@@ -236,17 +241,19 @@ def get_exception_message(ctx, script):
...
-                get_exception_message(ctx, 'oops()'))
+                (
+                    u'my_file_áéíóú.js:2\n'
+                    u'      thereMayBeErrors();\n'
+                    u'      ^\n'
+                    u'ReferenceError: thereMayBeErrors is not defined\n'
+                    u'    at oops (my_file_áéíóú.js:2:3)\n'
+                    u'    at <anonymous>:1:1'
+                ).encode('utf-8'),
+                get_exception_message(ctx, 'oops()').encode('utf-8'))
```
Comment (nitely) [inline tests/tests_context.py]: Same as above, don't encode and don't add the `u` prefix.

**Comment by nitely** [comment]
> Great. Now we need to avoid installing cffi on PyPy and we're done.

@thedrow I have checked some other cffi libs and it seems no one does that. I think we could just require `cffi>=1.6.0` and add PyPy X (where X is whatever version that ships cffi +1.6) to the compatibility list, that way it won't be (re)installed.

**Comment by thedrow** [comment]
You can't require cffi on installation since it's not installable on PyPy. cffi is bundled with the PyPy version you have. If you'll try to install it you'll see errors.

**Comment by nitely** [comment]
If you require `cffi>=1.6.0` (notice the _greater-than_ sign) and PyPy ships with cffi +1.6 (ie: 1.6.x, 1.7.x, etc) then it won't be installed.

**Comment by thedrow** [comment]
But CFFI is not installed as a site package. It's a bultin module in PyPy.

**Comment by nitely** [comment]
~~Are you sure? Coz in PyPy3 it is~~ Nevermind, I had another cffi installed, it's a built-in, but still --> that's the way the cffi docs say you should distribute your package.

http://cffi.readthedocs.io/en/release-1.6/cdef.html

``` python
# setup.py (with automatic dependency tracking)
from setuptools import setup

setup(
    ...,
    setup_requires=["cffi>=1.0.0"],
    cffi_modules=["package/foo_build.py:ffi"],
    install_requires=["cffi>=1.0.0"],
)
```

**Comment by heston** [comment]
@nitely Updated. Thanks for taking a look!

**Comment by thedrow** [comment]
@heston Can we add PyPy 2 to the build? Preferably the latest.
See https://github.com/Parsely/pykafka/blob/master/.travis.yml#L34 for how.

**Comment by nitely** [comment]
@thedrow yes, but please create a new PR.

LGTM

**Pull request merged**